Tests/DatadogTests/Datadog/LoggerTests.swift

```diff
@@ -622,6 +622,150 @@ class LoggerTests: XCTestCase {
         }
     }
 
+    func testWhenSendingErrorOrCriticalLogsWithAttributes_itCreatesRUMErrorForCurrentViewWithAttributes() throws {
+        let logging: LoggingFeature = .mockAny()
+        core.register(feature: logging)
+
+        let rum: RUMFeature = .mockWith(messageReceiver: ErrorMessageReceiver())
+        core.register(feature: rum)
+
+        // given
+        let logger = Logger.builder.build(in: core)
+        Global.rum = RUMMonitor(
+            core: core,
+            dependencies: RUMScopeDependencies(
+                core: core,
+                rumFeature: rum
+            ).replacing(viewUpdatesThrottlerFactory: { NoOpRUMViewUpdatesThrottler() }),
+            dateProvider: SystemDateProvider()
+        )
+        Global.rum.startView(viewController: mockView)
+        defer { Global.rum = DDNoopRUMMonitor() }
+
+        // when
+        let attributeValueA: String = .mockRandom()
+        logger.error("error message", attributes: [
+            "any_attribute_a": attributeValueA
+        ])
+        let attributeValueB: String = .mockRandom()
+        logger.critical("critical message", attributes: [
+            "any_attribute_b": attributeValueB
+        ])
+
+        // then
+        let rumEventMatchers = try core.waitAndReturnRUMEventMatchers()
```

Comment:
nit/ We should rather use the new `core.waitAndReturnEvents(of:ofType:)` API [added with `DatadogCoreProxy`](https://github.com/DataDog/dd-sdk-ios/wiki/How-to-use-%60DatadogCoreProxy%60-in-tests%3F#how-to-use-datadogcoreproxy). It leads to much simpler assertions and does not throw any errors.